sdk/ai/ai-projects/src/agents/utils.ts

```diff
@@ -40,17 +40,154 @@ export enum connectionToolType {
 }
 
 /**
- * Creates a tool definition for a connection tool with the given connection ids.
+ * Creates a connection tool
  *
  * @param toolType - The type of the connection tool.
  * @param connectionIds - A list of the IDs of the connections to use.
- * @returns The function tool definition.
+ * @returns An object containing the definition for the connection tool
  */
-export function fromConnectionId(toolType : connectionToolType, connectionIds: string[]) : ToolDefinitionParent {
+export function createConnectionTool(
+  toolType : connectionToolType,
+  connectionIds: string[]
+) : { definition : ToolDefinition } {
   return {
-    type: toolType,
-    [toolType]: {
-      connections: connectionIds.map(connectionId => ({connection_id: connectionId}))
+    definition : {
+      type: toolType,
+      [toolType]: {
+        connections: connectionIds.map(connectionId => ({connection_id: connectionId}))
+      }
     }
   }
 }
+
+/**
+ * Creates a file search tool
+ *
+ * @param vectorStoreIds - The ID of the vector store attached to this agent. There can be a maximum of 1 vector store attached to the agent.
+ * @param vectorStores - The list of vector store configuration objects from Azure. This list is limited to one element. The only element of this list contains the list of azure asset IDs used by the search tool.
+ * @param definitionDetails - The input definition information for a file search tool as used to configure an agent.
+ * 
+ * @returns An object containing the definition and resources for the file search tool
+ */
+export function createFileSearchTool(
+  vectorStoreIds?: string[],
+  vectorStores?: Array<VectorStoreConfigurations>,
+  definitionDetails? : FileSearchToolDefinitionDetails
+) : { definition : FileSearchToolDefinition, resources: ToolResources} {
+  return {
+    definition : { type: "file_search", file_search: definitionDetails },
+    resources : { file_search: { vector_store_ids: vectorStoreIds, vector_stores: vectorStores} }
+  }
+}
+
+/**
+ * Creates a code interpreter tool
+ * 
+ * @param fileIds - A list of file IDs made available to the `code_interpreter` tool. There can be a maximum of 20 files associated with the tool.
+ * @param dataSources - The data sources to be used. This option is mutually exclusive with fileIds.
+ * 
+ * @returns An object containing the definition and resources for the code interpreter tool.
+ */
+export function createCodeInterpreterTool( 
```

Comment:
Expose these as static methods on ToolSet

Reply:
If we do this does that require that they instantiate a ToolSet to use them? And if that's the case is the difference between creating a toolset and adding a tool vs. just creating the tool (through one of these methods) too confusing?

Reply:
No need to initialize the toolset. To differentiate between ToolSet and these factory utilities either expose them in a toolUtility static class and consume it something like
`ToolUtility.createCodeInterpreter ()`